### PR TITLE
Paintings icons and persistence.

### PR DESCRIPTION
**Pull request by AnturK**
image

Restores canvases showing up their content on the map.
This only gets updated on finalizing/naming the painting.
Adds persistent paintings, set persistence_id on sign/painting on your map to use it.

Bumps rustg up to 4.3

DNM, because this is bit experimental and requires a testmerge.

### CODE REVIEW

**Comment by Mickyan** [comment]
Aw I was hoping for a canvas bag to store my paintings

Great job though!

**Comment by AnturK** [comment]
Alright, review ready now.

File: code/__DEFINES/rust_g.dm [hunk 1]
```diff
@@ -6,6 +6,7 @@
 #define RUSTG_JOB_ERROR "JOB PANICKED"
 
 #define rustg_dmi_strip_metadata(fname) call(RUST_G, "dmi_strip_metadata")(fname)
+#define rustg_dmi_create_png(fname,width,height,data) call(RUST_G, "dmi_create_png")(fname,width,height,data)
```
Comment (Qustinnus) [inline code/__DEFINES/rust_g.dm]: bit of a nitpick but can you link to the rustg PR in PRs that update rustg so I know what to look at?

Comment (AnturK) [reply code/__DEFINES/rust_g.dm]: https://github.com/tgstation/rust-g/pull/21
https://github.com/tgstation/rust-g/pull/22

**Comment by optimumtact** [comment]
compile?

**Comment by AnturK** [comment]
Travis error is just the icon state error.

**Pull request merged**